tiledb/sm/subarray/subarray.cc

```diff
@@ -154,13 +155,119 @@ Subarray& Subarray::operator=(Subarray&& subarray) noexcept {
 /*               API              */
 /* ****************************** */
 
+void Subarray::add_label_range(
+    const DimensionLabelReference& dim_label_ref,
+    Range&& range,
+    const bool read_range_oob_error) {
+  const auto dim_idx = dim_label_ref.dimension_id();
+  if (label_range_subset_[dim_idx].has_value()) {
+    // A label range has already been set on this dimension. Do the following:
+    //  * Check this label is the same label that rangers were already set.
+    if (dim_label_ref.name() != label_range_subset_[dim_idx].name())
+      throw StatusException(Status_SubarrayError(
```

Comment:
NIT: Can you remove one line ifs? We are trying to get rid of them in the whole code base as they are error prone.